File: src/services/userService.js
import api from './api';

export const userService = {
    // Jobs
    getAllJobs: () => api.get('/user/jobs'),
    getJobById: (id) => api.get(`/user/jobs/${id}`),
    searchJobs: (filters) => api.post('/user/jobs/search', filters),

    // Applications
    applyForJob: (data) => api.post('/user/applications', data),
    getMyApplications: () => api.get('/user/applications'),

    // Saved Jobs
    saveJob: (jobId) => api.post(`/user/jobs/${jobId}/save`),
    getSavedJobs: () => api.get('/user/saved-jobs'),
    unsaveJob: (jobId) => api.delete(`/user/jobs/${jobId}/save`),

    // Profile
    getProfile: () => api.get('/user/profile'),
    updateProfile: (data) => api.put('/user/profile', data),

    // Dashboard stats
    getDashboardStats: () => api.get('/user/dashboard'),
};

File: backend/src/main/java/com/luzo_portal/service/UserService.java
package com.luzo_portal.service;

import com.luzo_portal.dto.ApplicationRequest;
import com.luzo_portal.dto.ApplicationResponse;
import com.luzo_portal.dto.JobFilterRequest;
import com.luzo_portal.dto.JobResponse;
import com.luzo_portal.dto.UpdateProfileRequest;
import com.luzo_portal.dto.UserDashboardStats;
import com.luzo_portal.dto.UserResponse;

import java.util.List;

public interface UserService {

    List<JobResponse> searchJobs(JobFilterRequest filterRequest);

    JobResponse getJobById(Long jobId);

    void saveJob(Long jobId);

    void unsaveJob(Long jobId);

    List<JobResponse> getSavedJobs();

    ApplicationResponse applyForJob(ApplicationRequest request);

    List<ApplicationResponse> getMyApplications();

    List<JobResponse> getSmartRecommendations();

    UserDashboardStats getDashboardStats();

    UserResponse getProfile();

    UserResponse updateProfile(UpdateProfileRequest request);
}